stan/math/prim/scal/fun/acosh.hpp

```diff
@@ -32,7 +37,18 @@ inline double acosh(double x) {
  * @return Inverse hyperbolic cosine of the argument.
  * @throw std::domain_error If argument is less than 1.
  */
-inline double acosh(int x) { return acosh(static_cast<double>(x)); }
+inline double acosh(int x) {
+  if (is_nan(x)) {
+    return x;
+  } else {
+    check_greater_or_equal("acosh", "x", x, 1);
+#ifdef _WIN32
```

Comment:
Do you remember what this was about? Why do we need this `ifdef`?

Reply:
Yep. acosh(inf) is expected to return inf, but under windows acosh(inf) returns nan. More background here: https://github.com/stan-dev/math/pull/1067#issuecomment-444145018